### PR TITLE
Initial commit of fetch-git

### PR DESCRIPTION
**Pull request by grahamc**
Just what it says on the tin, check out ./fetch-git/README.md for some details!

### CODE REVIEW

File: fetch-git/src/fetch-git.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          "debug": log.debug_msgs
+      })
+    ))
+
+
+parser = argparse.ArgumentParser(
+    description="an improved evaluation-time Git fetcher for Nix",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog="""
+DIFFERENCES FROM builtins.fetchGit
```
Comment (shlevy) [inline fetch-git/src/fetch-git.py]: Can we move/copy the DIFFERENCES section to the README?

**Review by shlevy — APPROVED** [review]
:fire:

**Comment by grahamc** [comment]
:heavy_check_mark: added a test for annotated tags
:heavy_check_mark: fixed a flake8 complaint about an unnecessary import

File: fetch-git/src/fetch-git.py (hunk 1 above)
Comment (grahamc) [reply fetch-git/src/fetch-git.py]: Done

**Comment by shlevy** [comment]
:ship:

**Comment by dtzWill** [comment]
Woohoo!

**Pull request merged**